keras/engine/training.py

```diff
@@ -3753,6 +3780,28 @@ def _should_eval(self, epoch, validation_freq):
                 f"type {type(validation_freq)}."
             )
 
+    @property
+    def states(self):
+        """Returns the list of all layer states.
+
+        Returns:
+          A list of variables.
+        """
+        states = []
+        # needs to be recursive, as Bidirectional don't have stateful argument
+        for l in self._flatten_layers(include_self=False, recursive=True):
+            if getattr(l, "stateful", False):
+                states += l.states
+        return self._dedup_weights(states)
+
+    def get_states(self):
```

Comment:
I think this method is not necessary.

Reply:
I added it as equivalent to the `get_weights()` method. Because `states` gives you a list of `tf.Variable`. But for `reset_states(...)` you need a list of `np.ndarray`. And this is what `get_states()` returns.